### DISCUSSION TITLE
adding 2D coordinates changes InChI string

### DISCUSSION THREAD

**Discussion by jasondbiggs** [rdkit/rdkit#3960] [Q&A] [answered]
If I create a molecule with undefined double-bond stereo and ask for its InChI string I get one result
```
In[56]:=
inchi.MolToInchi(MolFromSmiles('CC=C(C)O'))

Out[56]= "InChI=1S/C4H8O/c1-3-4(2)5/h3,5H,1-2H3"
```
but if I embed the molecule in 2D then I get a different result
```
In[57]:=
from rdkit.Chem import rdDepictor
m = MolFromSmiles('CC=C(C)O')
rdDepictor.Compute2DCoords(m)
inchi.MolToInchi(m)

Out[57]= "InChI=1S/C4H8O/c1-3-4(2)5/h3,5H,1-2H3/b4-3-"
```

This seems to be baked into the InChI software, that if 2D coordinates are available then it uses them to assign double-bond stereo.  Am I alone in thinking this is undesirable?  I think I need to start removing conformers before getting an InChI string now, and then adding it back afterwards.

**Comment by greglandrum** [comment, marked as answer]
The problem here is that the coordinates of the molecule define the stereochemistry of double bonds. Unless you explicitly tell it to do otherwise, InChI uses that definition.
You currently need to mark double bonds as being STEREOANY in order to work around this:
```
In [24]: m = Chem.MolFromSmiles('CC=C(C)O')

In [25]: rdDepictor.Compute2DCoords(m)
Out[25]: 0

In [26]: Chem.MolToInchi(m)
Out[26]: 'InChI=1S/C4H8O/c1-3-4(2)5/h3,5H,1-2H3/b4-3-'

In [27]: m.GetBondWithIdx(1).SetStereo(Chem.BondStereo.STEREOANY)

In [28]: Chem.MolToInchi(m)
[17:17:10] WARNING: Omitted undefined stereo
Out[28]: 'InChI=1S/C4H8O/c1-3-4(2)5/h3,5H,1-2H3'
```
A more general approach to this could be:
```
In [29]: m = Chem.MolFromSmiles('CC=C(C)O')

In [30]: rdDepictor.Compute2DCoords(m)
Out[30]: 0

In [31]: for bond in m.GetBonds():
    ...:     if bond.GetBondType()==Chem.BondType.DOUBLE and bond.GetStereo()==Chem.BondStereo.STEREONONE:
    ...:         bond.SetStereo(Chem.BondStereo.STEREOANY)
    ...: 

In [32]: Chem.MolToInchi(m)
[17:20:16] WARNING: Omitted undefined stereo
Out[32]: 'InChI=1S/C4H8O/c1-3-4(2)5/h3,5H,1-2H3'
```
The simplest alternative is to take advantage of the fact that `Chem.MolToMolBlock()` already does this for you and call the InChI code like this:
```
n [33]: m = Chem.MolFromSmiles('CC=C(C)O')

In [34]: rdDepictor.Compute2DCoords(m)
Out[34]: 0

In [35]: Chem.MolBlockToInchi(Chem.MolToMolBlock(m))
[17:21:32] WARNING: Omitted undefined stereo
Out[35]: 'InChI=1S/C4H8O/c1-3-4(2)5/h3,5H,1-2H3'
```

**Reply by jasondbiggs** [reply]
That works for me - thanks
